spec/index.adoc

```diff
@@ -1654,6 +1654,28 @@ There is no restriction on who can invoke this method.
 
 This method takes no input and returns 32 pseudo-random bytes to the caller. The return value is unknown to any part of the IC at time of the submission of this call. A new return value is generated for each call to this method.
 
+[#ic-get_ecdsa_public_key]
+=== IC method `get_ecdsa_public_key`
+
+This method returns the threshold ECDSA public key for the given canister using the given derivation path. If the `canister_id` is unspecified, it will default to the canister id of the caller.
+The `derivation_path` is a vector of variable length byte strings.
+For https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki[BIP-0032] compatibility, each byte string (`blob`) must be a 4-byte big-endian encoding of an unsigned integer less than 2^31^ (non-hardened).
+The only supported `key_id` value at the moment is `"secp256k1"`.
+
+The return result also contains a `chain_code` which can be used for key derivation purpose if needed.
+
+This call requires that the ECDSA feature is enabled and working properly, and the `canister_id` meets the requirement of a canister id.
+Otherwise it will be rejected.
```

Comment:
I wonder if we should really make this clarification, or just defer the publication of the spec to when the feature is enabled.

Reply:
If this is supposed to be _temporal_, then indeed we should take care using the versioning of the interface spec. (Each replica used on a given subnet implements some specified version of the interface spec.) If it's about being enabled on a subnet (so we'd switch the feature on and off per subnet), then I think the comment is appropriate. Do we plan the latter (except for using different versions during the roll-out)?